docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,290 @@
+from dataclasses import dataclass, field
+import uuid
+import itertools
+from typing import (
+    Generator,
+    Iterator,
+    Dict,
+    Sequence,
+    Optional,
+    TYPE_CHECKING,
+    Union,
+    Tuple,
+    List,
+)
+
+import weaviate
+import uuid
+import scipy.sparse
+
+from .... import Document
+from ..base.backend import BaseBackendMixin
+
+if TYPE_CHECKING:
+    from ....types import (
+        DocumentArraySourceType,
+    )
+
+
+@dataclass
+class WeaviateConfig:
+    """This class stores the config variables to initialize
+    connection to the Weaviate server"""
+
+    client: Optional[Union[str, weaviate.Client]] = None
+    n_dim: Optional[int] = None
+    array_id: Optional[str] = None
+
+
+class BackendMixin(BaseBackendMixin):
+    """Provide necessary functions to enable this storage backend. """
+
+    def _init_storage(
+        self,
+        docs: Optional['DocumentArraySourceType'] = None,
+        config: Optional[WeaviateConfig] = None,
+    ):
+        """Initialize weaviate storage.
+
+        :param docs: the list of documents to initialize to
+        :param config: the config object used to ininitialize connection to weaviate server
+        :raises ValueError: only one of array_id or docs can be used for initialization,
+            raise an error if both are provided
+        """
+
+        from ... import DocumentArray
+
+        self._schemas = None
+
+        if not config:
+            config = WeaviateConfig()
+
+        self.n_dim = config.n_dim or 1
+
+        if config.client is None:
+            self._client = weaviate.Client('http://localhost:8080')
+        elif isinstance(config.client, str):
+            self._client = weaviate.Client(config.client)
+        else:
+            self._client = config.client
+
+        if config.array_id is not None and docs is not None:
+            raise ValueError(
+                'only one of array_id or docs can be provided for initialization'
```

Comment:
what's the reason for this constraint?

Reply:
hm I think if the given `array_id` has non-empty data, then it might be hard to decide what data this DocArray should be initialized to? I think if we extend `docs` to an already existing data given by `array_id` then we are not really initializing to either but performing a new `extend`?

Reply:
in this case, the given DocArray could be appended into the `array_id` collection?

Reply:
As discussed offline, will align with Sqlite behavior but will also raise this for more feedback